### ISSUE TITLE
axie_tflite not working on S22 Exynos: libc++abi: terminating due to uncaught exception of type std::invalid_argument: stoi: no conversion 

### ISSUE THREAD

**Issue by v-prgmr** [argmaxinc/WhisperKitAndroid#2]
After building the axie_tflite binary from within the docker container and pushing the libs and binary to S22 with Exynos, the following issue occurs while executing the axie_tflite binary

![image](https://github.com/user-attachments/assets/153111cb-7fa9-4c53-8e31-9b4b58299e51)

**Comment by v-prgmr** [comment]
here is a dump of the logcat
`10-24 16:52:20.589  1376  2552 D RestrictionPolicy: isCameraEnabled ret(true) userId(0) cxtInfo.mCallerUid(1000) cxtInfo.mContainerId(0)
10-24 16:52:20.811  4758  4758 I tflite  : Initialized TensorFlow Lite runtime.
10-24 16:52:20.817  4758  4758 E libc++abi: terminating due to uncaught exception of type std::invalid_argument: stoi: no conversion
--------- beginning of crash
10-24 16:52:20.819  4758  4758 F libc    : Fatal signal 6 (SIGABRT), code -1 (SI_QUEUE) in tid 4758 (axie_tflite), pid 4758 (axie_tflite)
10-24 16:52:20.886  4761  4761 I crash_dump64: obtaining output fd from tombstoned, type: kDebuggerdTombstoneProto
10-24 16:52:20.891   728   728 I tombstoned: received crash request for pid 4758
10-24 16:52:20.893  4761  4761 I crash_dump64: performing dump of process 4758 (target tid = 4758)
10-24 16:52:20.907   904   904 E audit   : type=1400 audit(1729781540.905:15634): avc:  denied  { search } for  pid=4761 comm="crash_dump64" name="0" dev="tmpfs" ino=15 scontext=u:r:crash_dump:s0 tcontext=u:object_r:mnt_user_file:s0 tclass=dir permissive=0 SEPF_SM-S901B_12_0001 audit_filtered
10-24 16:52:20.908   904   904 E audit   : type=1400 audit(1729781540.905:15635): avc:  denied  { search } for  pid=4761 comm="crash_dump64" name="0" dev="tmpfs" ino=15 scontext=u:r:crash_dump:s0 tcontext=u:object_r:mnt_user_file:s0 tclass=dir permissive=0 SEPF_SM-S901B_12_0001 audit_filtered
10-24 16:52:20.948  4761  4761 F DEBUG   : *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
10-24 16:52:20.949  4761  4761 F DEBUG   : Build fingerprint: 'samsung/r0sxeea/r0s:14/UP1A.231005.007/S901BXXSBEXG8:user/release-keys'
10-24 16:52:20.949  4761  4761 F DEBUG   : Revision: '28'
10-24 16:52:20.949  4761  4761 F DEBUG   : ABI: 'arm64'
10-24 16:52:20.949  4761  4761 F DEBUG   : Processor: '3'
10-24 16:52:20.949  4761  4761 F DEBUG   : Timestamp: 2024-10-24 16:52:20.896465040+0200
10-24 16:52:20.949  4761  4761 F DEBUG   : Process uptime: 1s
10-24 16:52:20.949  4761  4761 F DEBUG   : Cmdline: axie_tflite /sdcard/argmax/tflite/inputs/jfk.flac tiny
10-24 16:52:20.949  4761  4761 F DEBUG   : pid: 4758, tid: 4758, name: axie_tflite  >>> axie_tflite <<<
10-24 16:52:20.949  4761  4761 F DEBUG   : uid: 2000
10-24 16:52:20.949  4761  4761 F DEBUG   : tagged_addr_ctrl: 0000000000000001 (PR_TAGGED_ADDR_ENABLE)
10-24 16:52:20.949  4761  4761 F DEBUG   : pac_enabled_keys: 000000000000000f (PR_PAC_APIAKEY, PR_PAC_APIBKEY, PR_PAC_APDAKEY, PR_PAC_APDBKEY)
10-24 16:52:20.949  4761  4761 F DEBUG   : signal 6 (SIGABRT), code -1 (SI_QUEUE), fault addr --------
10-24 16:52:20.949  4761  4761 F DEBUG   : Abort message: 'terminating due to uncaught exception of type std::invalid_argument: stoi: no conversion'
10-24 16:52:20.949  4761  4761 F DEBUG   :     x0  0000000000000000  x1  0000000000001296  x2  0000000000000006  x3  0000007fcb585b70
10-24 16:52:20.949  4761  4761 F DEBUG   :     x4  646277641f736766  x5  646277641f736766  x6  646277641f736766  x7  7f7f7f7f7f7f7f7f
10-24 16:52:20.949  4761  4761 F DEBUG   :     x8  00000000000000f0  x9  00000076546391e8  x10 0000000000000001  x11 000000765467f1f0
10-24 16:52:20.949  4761  4761 F DEBUG   :     x12 000000765b249020  x13 000000007fffffff  x14 000000000075796c  x15 000000ac0d627240
10-24 16:52:20.949  4761  4761 F DEBUG   :     x16 00000076546e3d08  x17 00000076546c10b0  x18 000000765b06a000  x19 0000000000001296
10-24 16:52:20.949  4761  4761 F DEBUG   :     x20 0000000000001296  x21 00000000ffffffff  x22 0000007fcb585ca0  x23 0000007fcb585ce0
10-24 16:52:20.949  4761  4761 F DEBUG   :     x24 0000007fcb585d90  x25 0000007659f67000  x26 0000007659f67000  x27 0000007659f67000
10-24 16:52:20.949  4761  4761 F DEBUG   :     x28 0000000000000000  x29 0000007fcb585bf0
10-24 16:52:20.949  4761  4761 F DEBUG   :     lr  0000007654670b4c  sp  0000007fcb585b50  pc  0000007654670b78  pst 0000000000001000
10-24 16:52:20.949  4761  4761 F DEBUG   : 19 total frames
10-24 16:52:20.949  4761  4761 F DEBUG   : backtrace:
10-24 16:52:20.949  4761  4761 F DEBUG   :   NOTE: Function names and BuildId information is missing for some frames due
10-24 16:52:20.949  4761  4761 F DEBUG   :   NOTE: to unreadable libraries. For unwinds of apps, only shared libraries
10-24 16:52:20.949  4761  4761 F DEBUG   :   NOTE: found under the lib/ directory are readable.
10-24 16:52:20.949  4761  4761 F DEBUG   :   NOTE: On this device, run setenforce 0 to make the libraries readable.
10-24 16:52:20.949  4761  4761 F DEBUG   :   NOTE: Unreadable libraries:
10-24 16:52:20.949  4761  4761 F DEBUG   :   NOTE:   /data/local/tmp/bin/axie_tflite
10-24 16:52:20.949  4761  4761 F DEBUG   :   NOTE:   /data/local/tmp/lib/libQnnTFLiteDelegate.so
10-24 16:52:20.949  4761  4761 F DEBUG   :       #00 pc 0000000000059b78  /apex/com.android.runtime/lib64/bionic/libc.so (abort+164) (BuildId: 7b2771e16ba279a5186fe9e8c815e964)
10-24 16:52:20.949  4761  4761 F DEBUG   :       #01 pc 00000000000cb128  /data/local/tmp/lib/libQnnTFLiteDelegate.so
10-24 16:52:20.949  4761  4761 F DEBUG   :       #02 pc 00000000000cacf4  /data/local/tmp/lib/libQnnTFLiteDelegate.so
10-24 16:52:20.949  4761  4761 F DEBUG   :       #03 pc 00000000000cabb0  /data/local/tmp/lib/libQnnTFLiteDelegate.so
10-24 16:52:20.949  4761  4761 F DEBUG   :       #04 pc 00000000000e31a4  /data/local/tmp/lib/libQnnTFLiteDelegate.so
10-24 16:52:20.949  4761  4761 F DEBUG   :       #05 pc 00000000000e3118  /data/local/tmp/lib/libQnnTFLiteDelegate.so
10-24 16:52:20.949  4761  4761 F DEBUG   :       #06 pc 00000000000a68f4  /data/local/tmp/lib/libQnnTFLiteDelegate.so
10-24 16:52:20.949  4761  4761 F DEBUG   :       #07 pc 00000000000a60f8  /data/local/tmp/lib/libQnnTFLiteDelegate.so
10-24 16:52:20.949  4761  4761 F DEBUG   :       #08 pc 000000000008a914  /data/local/tmp/lib/libQnnTFLiteDelegate.so
10-24 16:52:20.949  4761  4761 F DEBUG   :       #09 pc 000000000008ad38  /data/local/tmp/lib/libQnnTFLiteDelegate.so
10-24 16:52:20.949  4761  4761 F DEBUG   :       #10 pc 00000000000523a8  /data/local/tmp/lib/libQnnTFLiteDelegate.so
10-24 16:52:20.949  4761  4761 F DEBUG   :       #11 pc 0000000000051fcc  /data/local/tmp/lib/libQnnTFLiteDelegate.so
10-24 16:52:20.949  4761  4761 F DEBUG   :       #12 pc 0000000000117ec0  /data/local/tmp/bin/axie_tflite
10-24 16:52:20.949  4761  4761 F DEBUG   :       #13 pc 0000000000118614  /data/local/tmp/bin/axie_tflite
10-24 16:52:20.949  4761  4761 F DEBUG   :       #14 pc 00000000000efdcc  /data/local/tmp/bin/axie_tflite
10-24 16:52:20.949  4761  4761 F DEBUG   :       #15 pc 00000000000d576c  /data/local/tmp/bin/axie_tflite
10-24 16:52:20.949  4761  4761 F DEBUG   :       #16 pc 00000000000e7b64  /data/local/tmp/bin/axie_tflite
10-24 16:52:20.949  4761  4761 F DEBUG   :       #17 pc 00000000000e94f0  /data/local/tmp/bin/axie_tflite
10-24 16:52:20.949  4761  4761 F DEBUG   :       #18 pc 00000000000526d8  /apex/com.android.runtime/lib64/bionic/libc.so (__libc_init+104) (BuildId: 7b2771e16ba279a5186fe9e8c815e964)`

**Comment by keith4ever** [comment]
Hello,

Thanks for evaluating whisperkit. 
Currently only .wav format audio is supported as input, with 16Khz / mono format. 
We'll soon add various formats of audio codecs and sampling freq.

**Comment by v-prgmr** [comment]
@keith4ever Thank you for getting back, but I could do inference on S24 Ultra with the same binary and libs and flac audio as input. What am I missing here?

**Comment by v-prgmr** [comment]
@keith4ever right, so I tried passing in a .wav file, still have the same error on S22 Exynos. As you have mentioned, when I passed the .flac as input in the S24 Ultra with Qualcomm, the decoder outputs were wrong, but now even with feeding in inputs at 16 kHz in .wav format, gives wrong decoder outputs.

**Comment by v-prgmr** [comment]
@keith4ever you are right, I did not bother to look at the decoding results while loading in the .flac file. With audio as .wav input recorded at 16 kHz with mono channel, the decoder output looks correct. 

But still the error occurs on S22 Exynos irrespective of the audio format.

**Comment by atiorh** [comment]
I am adding S22 Exynos to our test device fleet ahead of our upcoming beta release. We will aim to explicitly support Exynos chips with the beta release and publish that as part of our device coverage documentation. Thanks for the report!

**Comment by keith4ever** [comment]
Hi @v-prgmr ,
As of Dec 24th, we have added other audio code support and it should work in your env. now. 
please check out the latest code and see if it works.

**Comment by bpkeene** [comment]
Closing as this is stale
